src/eth_provider/provider.rs

```diff
@@ -939,3 +956,56 @@ where
         Ok(())
     }
 }
+
+impl<SP> EthDataProvider<SP>
+where
+    SP: starknet::providers::Provider + Send + Sync,
+{
+    pub async fn retry_transactions(&self) -> EthProviderResult<Vec<B256>> {
+        // Initialize an empty vector to store the hashes of retried transactions
+        let mut transactions_retried = Vec::new();
+
+        // Iterate over pending transactions fetched from the database
+        for tx in self.database.get::<StoredPendingTransaction>(None, None).await? {
```

Comment:
I wonder if we should have a window of blocks for which we will retry? Otherwise this query could start getting big

Reply:
I wondered at the start but as we gradually delete the transactions from the pending pool, I would say that this request still remains reasonable, right?

Reply:
ah right we also remove them, ok